### PR TITLE
#61 Rename domain objects to be factories

### LINKED ISSUE #61: Refactor the domain objects to be true factories / generators
Issue Description
---
The domain objects are not true domain objects but are domain object factories (or generators) that create dictionaries containing the domain object fields. 

Decide on behaviour and naming convention: Are these factories or generators? 
Rename the domain objects appropriately to show that they are factories/ generators.
Add domain object specific config (normal and custom) to the factory/generator so that these can be passed around instead of the config itself. 

Design
---
Refactor to include the custom and normal config in the class as a dictionary. 

Refactor parts of the code that use the domain objects and the config to use just the factory/generator

Documentation Changes
---
Update class documentation

Test Evidence
---
No new tests required but make sure existing tests are refactored to work

Validation in Develop
---

### PR DESCRIPTION
**Pull request by GalateaTom**
The Domain Object classes have been renamed to make it clear that they're factories.  I have also replaced uses of the word generate with create except in those circumstances when a generator/yield is actually being used.
Closes #61 
app.py still executes correctly both with and without the config files being stated.  The tests also still execute correctly.

### CODE REVIEW

File: src/domainobjects/counterparty_factory.py [hunk 1]
```diff
@@ -33,12 +33,12 @@ def generate(self, record_count, start_id):
         return records
 
     def get_record(self, current_id):
```
Comment (GalaJoe) [inline src/domainobjects/counterparty_factory.py]: Rename this method to create record for uniformity

File: tests/utils/helper_methods.py [hunk 2]
```diff
@@ -45,153 +45,153 @@ def query_db(table_name, attribute=None):
...
 
 
 def set_up_swap_contract_tests():
     delete_local_database()
-    generate_counterparty(20)
-    records = generate_swap_contract(20)
-    domain_obj = swap_contract.SwapContract(None, None)
+    create_counterparty(20)
+    records = create_swap_contract(20)
+    domain_obj = swap_contract_factory.SwapContractFactory(None, None)
```
Comment (GalaJoe) [inline tests/utils/helper_methods.py]: "domain_obj_factory ="

Comment (GalaJoe) [reply tests/utils/helper_methods.py]: Also in other test modules

File: tests/utils/helper_methods.py [hunk 3]
```diff
@@ -4,10 +4,10 @@
 
 sys.path.insert(0, 'src/')
 from database.sqlite_database import Sqlite_Database
-from domainobjects import back_office_position, cash_balance, cashflow
-from domainobjects import counterparty, depot_position, front_office_position
-from domainobjects import instrument, order_execution, price
-from domainobjects import stock_loan_position, swap_contract, swap_position
+from domainobjects import back_office_position_factory, cash_balance_factory, cashflow_factory
```
Comment (GalaJoe) [inline tests/utils/helper_methods.py]: From domain_factories, domain_object_factories? Not sure which is better.

File: tests/utils/helper_methods.py [hunk 4]
```diff
@@ -45,153 +45,153 @@ def query_db(table_name, attribute=None):
 
 
 # Generation Methods
-def generate_back_office_position(amount=1):
-    obj = back_office_position.BackOfficePosition(None, None)
-    return obj.generate(amount, 0)
+def create_back_office_position(amount=1):
+    obj = back_office_position_factory.BackOfficePositionFactory(None, None)
```
Comment (GalaJoe) [inline tests/utils/helper_methods.py:54]: Lots of obj, these are actually instantiated factories

**Review by GalaJoe — CHANGES_REQUESTED** [review]

**Comment by GalateaTom** [comment]
PR will be cancelled and re-raised as Hound isn't picking up all of the fixes

**Pull request closed without merging**